cmd/lakefs/cmd/kv.go

```diff
@@ -0,0 +1,103 @@
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"github.com/treeverse/lakefs/pkg/kv"
+	"github.com/treeverse/lakefs/pkg/logging"
+	"github.com/treeverse/lakefs/pkg/version"
+)
+
+const (
+	ScanCmdMinArgs = 1
+	ScanCmdMaxArgs = 2
+	GetCmdNumArgs  = 2
+)
+
+var kvCmd = &cobra.Command{
+	Use:    "kv",
+	Short:  "Inspect lakeFS' Key-Value Store",
+	Hidden: true,
+}
+
+var kvGetCmd = &cobra.Command{
+	Use:    "get <partition key> <key>",
+	Short:  "Return the value for the given key under the given partition",
+	Hidden: true,
+	Args:   cobra.ExactArgs(GetCmdNumArgs),
+	Run: func(cmd *cobra.Command, args []string) {
+		logger := logging.Default()
+		cfg := loadConfig()
+
+		ctx := cmd.Context()
+		logger.WithField("version", version.Version).Info("lakeFS kv")
+
+		kvParams := cfg.GetKVParams()
+		kvStore, err := kv.Open(ctx, kvParams)
+		if err != nil {
+			logger.WithError(err).Fatal("Failed to open KV store")
+		}
+		defer kvStore.Close()
+
+		val, err := kvStore.Get(ctx, []byte(args[0]), []byte(args[1]))
+		if err != nil {
+			logger.WithError(err).Fatal("Failed to get value")
+		}
+		prettyVal, err := kv.ToPrettyString(args[1], val.Value)
+		if err != nil {
+			logger.WithError(err).Fatal("Failed to build object from KV value")
+		}
+
+		fmt.Printf("%s:\n%s\n", args[1], prettyVal)
+	},
+}
+
+var kvScanCmd = &cobra.Command{
+	Use:    "scan <partition key> [<key>]",
+	Short:  "Scan through keys and values under the given partition. An optional key cna be specified as a starting point (inclusive)",
+	Hidden: true,
+	Args:   cobra.RangeArgs(ScanCmdMinArgs, ScanCmdMaxArgs),
+	Run: func(cmd *cobra.Command, args []string) {
+		logger := logging.Default()
+		cfg := loadConfig()
+
+		ctx := cmd.Context()
+		logger.WithField("version", version.Version).Info("lakeFS scan")
+
+		kvParams := cfg.GetKVParams()
+		kvStore, err := kv.Open(ctx, kvParams)
+		if err != nil {
+			logger.WithError(err).Fatal("Failed to open KV store")
+		}
+		defer kvStore.Close()
+
+		var start []byte = nil
+		if len(args) > ScanCmdMinArgs {
+			start = []byte(args[1])
+		}
+		iter, err := kvStore.Scan(ctx, []byte(args[0]), start)
+		if err != nil {
+			logger.WithError(err).Fatal("Scan failed")
```

Comment:
Relevant for couple of cases in this pr - we should print out the error and exit gracefully by return as we like to call out defer functions.

Reply:
Fixed